modules/imgproc/src/warp_kernels.simd.hpp

```diff
@@ -2130,6 +1592,1414 @@ void warpAffineLinearApproxInvoker_8UC4(const uint8_t *src_data, size_t src_step
+                } else {
+                    CV_WARP_LINEAR_VECTOR_SHUFFLE_NOTALLWITHIN(C3, 16U);
+                }
+
+                CV_WARP_LINEAR_VECTOR_INTER_LOAD_U16(C3);
+
+                CV_WARP_LINEAR_VECTOR_INTER_LOAD_U16F32(C3);
+
+                CV_WARP_LINEAR_VECTOR_INTER_CALC_F32(C3);
+
+                CV_WARP_LINEAR_VECTOR_INTER_STORE_F32U16(C3);
+            }
+#endif // (CV_SIMD || CV_SIMD_SCALABLE)
+
+            for (; x < dstcols; x++) {
+                float w = x*M[6] + y*M[7] + M[8];
+                float sx = (x*M[0] + y*M[1] + M[2]) / w;
+                float sy = (x*M[3] + y*M[4] + M[5]) / w;
+                int ix = cvFloor(sx), iy = cvFloor(sy);
+                sx -= ix; sy -= iy;
+                int p00r, p00g, p00b, p01r, p01g, p01b;
+                int p10r, p10g, p10b, p11r, p11g, p11b;
+                const uint16_t *srcptr = src + srcstep * iy + ix*3;
+
+                CV_WARP_LINEAR_SCALAR_SHUFFLE(C3);
+
+                CV_WARP_LINEAR_SCALAR_INTER_CALC_F32(C3);
+
+                CV_WARP_LINEAR_SCALAR_STORE(C3, 16U);
+            }
+        }
+    };
+    parallel_for_(Range(0, dst_rows), worker);
+}
+
+void warpPerspectiveLinearInvoker_16UC4(const uint16_t *src_data, size_t src_step, int src_rows, int src_cols,
+                                        uint16_t *dst_data, size_t dst_step, int dst_rows, int dst_cols,
+                                        const double dM[9], int border_type, const double border_value[4]) {
+    auto worker = [&](const Range &r) {
+        CV_INSTRUMENT_REGION();
+
+        const auto *src = src_data;
+        auto *dst = dst_data;
+        size_t srcstep = src_step/sizeof(uint16_t), dststep = dst_step/sizeof(uint16_t);
+        int srccols = src_cols, srcrows = src_rows;
+        int dstcols = dst_cols;
+        float M[9];
+        for (int i = 0; i < 9; i++) {
+            M[i] = static_cast<float>(dM[i]);
+        }
+        uint16_t bval[] = {
+            saturate_cast<uint16_t>(border_value[0]),
+            saturate_cast<uint16_t>(border_value[1]),
+            saturate_cast<uint16_t>(border_value[2]),
+            saturate_cast<uint16_t>(border_value[3]),
+        };
+        int border_type_x = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srccols <= 1 ? BORDER_REPLICATE : border_type;
+        int border_type_y = border_type != BORDER_CONSTANT &&
+                            border_type != BORDER_TRANSPARENT &&
+                            srcrows <= 1 ? BORDER_REPLICATE : border_type;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+        constexpr int max_vlanes_32{VTraits<v_float32>::max_nlanes};
+        constexpr int max_uf{max_vlanes_32*2};
+        int vlanes_32 = VTraits<v_float32>::vlanes();
+        // unrolling_factor = lane_size / 16 = vlanes_32 * 32 / 16 = vlanes_32 * 2
+        int uf = vlanes_32 * 2;
+
+        std::array<float, max_vlanes_32> start_indices;
```

Comment:
initialize start indices in registers first using vx_setzero and then store them to buffer. The use of std::array looks a bit strange since the rest of the code does not use std::

Reply:
> initialize start indices in registers first using vx_setzero and then store them to buffer.

I don't quite get the idea here. How do I make a incremental sequence {0, 1, 2, 3, ..., max_vlanes_32} using universal intrinsics?